gc/gc-base/src/main/java/org/projectnessie/gc/base/IdentifyContentsPerExecutor.java

```diff
@@ -284,27 +331,42 @@ private Iterator<Row> walkAllCommitsInReference(
                           .refName(Detached.REF_NAME)
                           .fetch(FetchOption.ALL),
                   OptionalInt.empty())
-              .filter(unprotectedCommitsPredicate)
-              .map(LogResponse.LogEntry::getOperations)
+              .map(
+                  entry -> {
+                    String hash = entry.getCommitMeta().getHash();
+                    commitHash.set(hash);
+                    commitTime.set(entry.getCommitMeta().getCommitTime());
+                    if (!foundCheckPoint.get()) {
+                      foundCheckPoint.set(
+                          commitCheckPoint != null && commitCheckPoint.equals(hash));
+                    }
+                    return entry.getOperations();
+                  })
               .flatMap(operations -> operations.stream().filter(Operation.Put.class::isInstance))
               .map(Operation.Put.class::cast)
               .map(Operation.Put::getContent)
               .filter(
                   content ->
                       (content.getType() == Content.Type.ICEBERG_TABLE
                           || content.getType() == Content.Type.ICEBERG_VIEW))
-              .filter(expiredContentPredicate)
+              .filter(validSnapshotPredicate)
               .iterator();
```

Comment:
An Iterator is most likely wrong here. Read up about terminal operations of Streams.

Reply:
`Iterable` bears the expectation to be able to provide an `Iterator` more than once, which a `Stream` can’t fulfill.
But I don't need `Iterable`, one time `Iterator` is enough here.